### DISCUSSION TITLE
Mixin transformation failed

### DISCUSSION THREAD

**Discussion by Intezium** [FabricMC/fabric-api#2395] [Mod Dev Support] [answered]
I need help.. [Log File.](https://pastebin.com/cmYj7k4R)
```
@Mixin(CraftingResultSlot.class)
public class CraftingResultSlotMixin {
    @Inject(at = @At(
            value = "INVOKE",
            target = "dropItem(Lnet/minecraft/item/ItemStack;Z)Lnet/minecraft/entity/ItemEntity;V"),
            method = "onTakeItem(Lnet/minecraft/entity/player/PlayerEntity;Lnet/minecraft/item/ItemStack;)V",
            cancellable = true)
    private void onTakeItem(final PlayerEntity player, final ItemStack itemStack, final CallbackInfoReturnable<Boolean> info) {
        ActionResult result = TakeCraftingItem.EVENT.invoker().interact(player, itemStack);

        if (result == ActionResult.FAIL) {
            info.cancel();
        }
    }
}
```

**Comment by warjort** [comment, marked as answer]
You are missing the class name for the dropItem and you seem to be trying to give it two return values? :-)
It should be more like this (I haven't tested it)
` target = "Lnet/minecraft/entity/player/PlayerEntity;dropItem(Lnet/minecraft/item/ItemStack;Z)Lnet/minecraft/entity/ItemEntity;"),`

**Reply by Intezium** [reply]
I checked your code, invalid descriptor. [Log](https://pastebin.com/r0KTYZf5)
I also copied the path to PlayerEntity from Intellij Idea.

**Reply by warjort** [reply]
Your method signature is wrong, it should be
`	    private void onTakeItem(final PlayerEntity player, final ItemStack itemStack, final CallbackInfo info) {`

You don't have access to a return value because you intercepting BEFORE the call to the method with INVOKE.

**Reply by Intezium** [reply]
Ohh my mind... It's really not obvious. Thank!
